packages/ag-charts-community/src/chart/interaction/consumableEvent.ts

```diff
@@ -12,12 +12,17 @@ type TypedListener<TType extends string, TEvent extends TypedConsumableEvent<TTy
     (event: TEvent) => void
 >;
 
-export function buildConsumable<T>(obj: T): T & ConsumableEvent {
+export function buildConsumable<T>(obj: T & { sourceEvent?: Event | ConsumableEvent }): T & ConsumableEvent {
```

Comment:
The change to this function is highly likely to break something with interactions.

The `consume()` function is intended to stop other listeners from receiving events. There are almost certainly some scenarios where we want to both call `preventDefault()` and have multiple listeners to do handle the event. So mixing up the behaviour of `consume()` / `preventDefault()` is not a good idea.

Instead, I think it would be better to have a simple solution like calling `preventDefault()` directly when an arrow key is pressed.

Reply:
I don't believe that we should have multiple listeners consuming the same consumable event - that is the whole point of it being a consumable event, it's only consumed once.

Calling `preventDefault()` for this very specific case (up/down arrows) is the wrong approach IMO, as it's a general principle that if we're consuming an event, the browsers default handling should not kick in.

Reply:
Discussed offline.